example-2/evm/src/contracts/lendingSpoke/Spoke.sol

```diff
@@ -4,33 +4,108 @@ pragma solidity ^0.8.0;
+            decimals: params.decimals,
+            exists: true
+        });
+        registerAssetInfo(params.assetAddress, info);        
+    }
+
+    function depositCollateral(address assetAddress, uint256 assetAmount) public {
+        PayloadHeader memory payloadHeader = PayloadHeader({
+            payloadID: 1,
+            sender: address(this)
+        });
+
+        DepositPayload memory depositPayload = DepositPayload({
+            header: payloadHeader,
+            assetAddress: assetAddress,
+            assetAmount: assetAmount
+        });
+
+        // create WH message
+        bytes memory serialized = encodeDepositPayload(depositPayload);
+
+        sendTokenBridgeMessage(assetAddress, assetAmount, serialized);
     }
 
-    function registerOnHub() public returns (uint64) {}
+    function withdrawCollateral(address assetAddress, uint256 assetAmount) public returns (uint64 sequence) {
+        PayloadHeader memory payloadHeader = PayloadHeader({
+            payloadID: 2,
+            sender: address(this)
+        });
 
-    function depositCollateral() public returns (uint64) {}
+        WithdrawPayload memory withdrawPayload = WithdrawPayload({
+            header: payloadHeader,
+            assetAddress: assetAddress,
+            assetAmount: assetAmount
+        });
 
-    function withdrawCollateral() public returns (uint64) {}
+        // create WH message
+        bytes memory serialized = encodeWithdrawPayload(withdrawPayload);
+
+        sequence = sendWormholeMessage(serialized);
+    }
 
-    function borrow() public returns (uint64) {}
+    function borrow(address assetAddress, uint256 assetAmount) public returns (uint64 sequence) {
+        PayloadHeader memory payloadHeader = PayloadHeader({
+            payloadID: 3,
+            sender: address(this)
+        });
 
-    function repay() public returns (uint64) {}
+        BorrowPayload memory borrowPayload = BorrowPayload({
+            header: payloadHeader,
+            assetAddress: assetAddress,
+            assetAmount: assetAmount
+        });
 
-    function liquidate(
-        address vault,
-        LiquidationRepay[] memory repay,
-        LiquidationReceipt[] memory receipt
-    ) public returns (uint64) {}
+        // create WH message
+        bytes memory serialized = encodeBorrowPayload(borrowPayload);
+
+        sequence = sendWormholeMessage(serialized);
+    }
+
+    function repay(address assetAddress, uint256 assetAmount) public {
+        PayloadHeader memory payloadHeader = PayloadHeader({
+            payloadID: 4,
+            sender: address(this)
+        });
+
+        RepayPayload memory repayPayload = RepayPayload({
+            header: payloadHeader,
+            assetAddress: assetAddress,
+            assetAmount: assetAmount
+        });
+
+        // create WH message
+        bytes memory serialized = encodeRepayPayload(repayPayload);
+
+        sendTokenBridgeMessage(assetAddress, assetAmount, serialized);
```

Comment:
again, want to send a bunch of tokens to the protocol. may not want to do multiple token mints for the reason that batch VAAs aren't fully out yet, but even for one token we want to send tokens to protocol before send WH msg